otsserver/rpc.py

```diff
@@ -154,42 +155,26 @@ def do_GET(self):
             # need to investigate further, but this seems to work.
             str_wallet_balance = str(proxy._call("getbalance"))
 
-            welcome_page = """\
-<html>
-<head>
-    <title>OpenTimestamps Calendar Server</title>
-</head>
-<body>
-<p>This is an <a href="https://opentimestamps.org">OpenTimestamps</a> <a href="https://github.com/opentimestamps/opentimestamps-server">Calendar Server</a> (v%s)</p>
-
-<p>
-Pending commitments: %d</br>
-Transactions waiting for confirmation: %d</br>
-Most recent timestamp tx: %s (%d prior versions)</br>
-Most recent merkle tree tip: %s</br>
-Best-block: %s, height %d</br>
-</br>
-Wallet balance: %s BTC</br>
-</p>
-
-<p>
-You can donate to the wallet by sending funds to: %s</br>
-This address changes after every donation.
-</p>
-
-</body>
-</html>
-""" % (otsserver.__version__,
-       len(self.calendar.stamper.pending_commitments),
-       len(self.calendar.stamper.txs_waiting_for_confirmation),
-       b2lx(self.calendar.stamper.unconfirmed_txs[-1].tx.GetTxid()) if self.calendar.stamper.unconfirmed_txs else 'None',
-       max(0, len(self.calendar.stamper.unconfirmed_txs) - 1),
-       b2x(self.calendar.stamper.unconfirmed_txs[-1].tip_timestamp.msg) if self.calendar.stamper.unconfirmed_txs else 'None',
-       bitcoin.core.b2lx(proxy.getbestblockhash()), proxy.getblockcount(),
-       str_wallet_balance,
-       str(proxy.getaccountaddress('')))
-
-            self.wfile.write(welcome_page.encode())
+            transactions = proxy._call("listtransactions", "", 50)
+            # We want only the confirmed txs containing an OP_RETURN, from most to least recent
+            transactions = list(filter(lambda x: x["confirmations"] > 0 and x["amount"] == 0, transactions))
+            transactions.sort(key=lambda x: x["confirmations"])
+
+            stats = { 'version': otsserver.__version__,
+              'pending_commitments': len(self.calendar.stamper.pending_commitments),
+              'txs_waiting_for_confirmation':len(self.calendar.stamper.txs_waiting_for_confirmation),
+              'most_recent_tx': b2lx(self.calendar.stamper.unconfirmed_txs[-1].tx.GetTxid()) if self.calendar.stamper.unconfirmed_txs else 'None',
+              'prior_versions': max(0, len(self.calendar.stamper.unconfirmed_txs) - 1),
+              'tip': b2x(self.calendar.stamper.unconfirmed_txs[-1].tip_timestamp.msg) if self.calendar.stamper.unconfirmed_txs else 'None',
+              'best_block': bitcoin.core.b2lx(proxy.getbestblockhash()),
+              'block_height': proxy.getblockcount(),
+              'balance': str_wallet_balance,
+              'address': str(proxy.getaccountaddress('')),
+              'transactions': transactions[:5],
+            }
+            welcome_page = renderer.render(open("otsserver/templates/index.mustache").read(), stats)
```

Comment:
This `open()` call assumes that the current directory is where otsserver is being run; this will conflict with other ways of installing `otsd`.

I'd suggest instead that we continue to have the template inline, as a multi-line comment.